packages/integration-tests/src/test-project-setup/passwordless_auth.ts

```diff
@@ -0,0 +1,92 @@
+import fs from 'fs/promises';
+import { CloudFormationClient } from '@aws-sdk/client-cloudformation';
+import { TestProjectBase } from './test_project_base.js';
+import { TestProjectCreator } from './test_project_creator.js';
+import { CognitoIdentityProviderClient } from '@aws-sdk/client-cognito-identity-provider';
+import { createEmptyAmplifyProject } from './create_empty_amplify_project.js';
+import { BackendIdentifier } from '@aws-amplify/plugin-types';
+import { DeployedResourcesFinder } from '../find_deployed_resource.js';
+import assert from 'node:assert';
+
+/**
+ * Creates test projects with passwordless auth.
+ */
+export class PasswordlessAuthTestProjectCreator implements TestProjectCreator {
+  readonly name = 'passwordless-auth';
+
+  /**
+   * Creates project creator.
+   */
+  constructor(
+    private readonly cfnClient: CloudFormationClient,
+    private readonly cognitoIdentityProviderClient: CognitoIdentityProviderClient,
```

Comment:
once this client is actually being used, we'll need to add some permissions to our e2e role policy